exercises/rational-numbers/.meta/src/reference/java/Rational.java

```diff
@@ -0,0 +1,102 @@
+public class Rational {
+    private int numerator;
+    private int denominator;
+
+    public Rational(final int numerator, final int denominator) throws IllegalArgumentException {
+        if (denominator == 0) {
+            throw new IllegalArgumentException("Denominator must not be zero.");
+        }
+
+        this.numerator = numerator;
+        this.denominator = denominator;
+
+        this.reduce();
+        this.ensureSignInNumerator();
+    }
+
+    private void reduce() {
+        final int commonDivisor = gcd(this.numerator, this.denominator);
+
+        this.numerator /= commonDivisor;
+        this.denominator /= commonDivisor;
+    }
+
+    private int gcd(int a, int b) {
+        while (b != 0) {
+            final int t = b;
+            b = a % b;
+            a = t;
+        }
+        return a;
+    }
+
+    private void ensureSignInNumerator() {
+        if (this.denominator < 0) {
+            this.denominator = -this.denominator;
+            this.numerator = -this.numerator;
+        }
+    }
+
+    public int getNumerator() {
+        return this.numerator;
+    }
+
+    public int getDenominator() {
+        return this.denominator;
+    }
+
+    public Rational abs() {
+        return new Rational(abs(this.numerator), abs(this.denominator));
+    }
+
+    private int abs(int x) {
+        final int mask = x >> 31;
+        return (mask ^ x) - mask;
+    }
+
+    public Rational add(Rational that) {
```

Comment:
There's a good few methods here that have a higher wider access modifier than they need to.

On the track here, we try to [adhere to best practise](https://github.com/exercism/java/blob/master/POLICIES.md#adhere-to-best-practices) as noted in our POLICIES.

This (and a few more) can be changed from `public` to package-private.